### PR TITLE
Add @var tag support in dockblocks

### PR DESCRIPTION
**Pull request by red-led**
Pretty common tag for class variables.

### CODE REVIEW

File: src/Generator/DocBlock/Tag/VarTag.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    public function getName()
+    {
+        return 'var';
+    }
+
+    /**
+     * @param string $variableName
+     * @return self
+     */
+    public function setVariableName($variableName)
```
Comment (Ocramius) [inline src/Generator/DocBlock/Tag/VarTag.php]: Please remove this setter, unless strictly required. Making it `private` is also OK

File: src/Generator/DocBlock/Tag/VarTag.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        $this->variableName = ltrim($variableName, '$');
+        return $this;
+    }
+
+    /**
+     * @return string
```
Comment (Ocramius) [inline src/Generator/DocBlock/Tag/VarTag.php]: May be `null`

File: src/Generator/DocBlock/Tag/VarTag.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ */
+
+namespace Zend\Code\Generator\DocBlock\Tag;
+
+class VarTag extends AbstractTypeableTag implements TagInterface
+{
+    /**
+     * @var string
+     */
+    protected $variableName = null;
```
Comment (Ocramius) [inline src/Generator/DocBlock/Tag/VarTag.php]: `= null` redundant

File: src/Generator/DocBlock/Tag/VarTag.php [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+ * @copyright Copyright (c) 2005-2016 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Code\Generator\DocBlock\Tag;
+
+class VarTag extends AbstractTypeableTag implements TagInterface
+{
+    /**
+     * @var string
```
Comment (Ocramius) [inline src/Generator/DocBlock/Tag/VarTag.php]: May be `null`, so should be `string|null`

File: src/Generator/DocBlock/Tag/VarTag.php [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    {
+        if (!empty($variableName)) {
+            $this->setVariableName($variableName);
+        }
+
+        parent::__construct($types, $description);
+    }
+
+    /**
+     * @return string
```
Comment (Ocramius) [inline src/Generator/DocBlock/Tag/VarTag.php]: `{@inheritDoc}`

File: src/Generator/DocBlock/Tag/VarTag.php [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+     */
+    public function getVariableName()
+    {
+        return $this->variableName;
+    }
+
+    /**
+     * @return string
```
Comment (Ocramius) [inline src/Generator/DocBlock/Tag/VarTag.php]: `{@inheritDoc}`

File: src/Generator/DocBlock/Tag/VarTag.php [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+     */
+    public function generate()
+    {
+        $output = '@var'
```
Comment (Ocramius) [inline src/Generator/DocBlock/Tag/VarTag.php]: Return directly

File: src/Generator/DocBlock/Tag/VarTag.php [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+     * @var string
+     */
+    protected $variableName = null;
+
+    /**
+     * @param string $variableName
+     * @param array $types
+     * @param string $description
+     */
+    public function __construct($variableName = null, $types = [], $description = null)
```
Comment (Ocramius) [inline src/Generator/DocBlock/Tag/VarTag.php]: is `$types` a `TypeGenerator[]` here? Can we enforce that?

File: src/Reflection/DocBlock/Tag/VarTag.php [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+    /**
+     * Initializer
+     *
+     * @param  string $tagDocblockLine
+     */
+    public function initialize($tagDocblockLine)
+    {
+        $match = [];
+        if (!preg_match('#^(.+)?(\$[\S]+)[\s]*(.*)$#m', $tagDocblockLine, $match)) {
```
Comment (Ocramius) [inline src/Reflection/DocBlock/Tag/VarTag.php]: the `[\s]*` can be simplified to `\s*`

File: src/Generator/DocBlock/Tag/VarTag.php (hunk 4 above)
Comment (red-led) [inline src/Generator/DocBlock/Tag/VarTag.php]: done

File: src/Generator/DocBlock/Tag/VarTag.php (hunk 3 above)
Comment (red-led) [inline src/Generator/DocBlock/Tag/VarTag.php]: done

File: src/Generator/DocBlock/Tag/VarTag.php (hunk 8 above)
Comment (red-led) [inline src/Generator/DocBlock/Tag/VarTag.php]: No, it is string or array of strings, passed to```\Zend\Code\Generator\DocBlock\Tag\AbstractTypeableTag::setTypes```

File: src/Generator/DocBlock/Tag/VarTag.php (hunk 5 above)
Comment (red-led) [inline src/Generator/DocBlock/Tag/VarTag.php]: done

File: src/Generator/DocBlock/Tag/VarTag.php (hunk 1 above)
Comment (red-led) [inline src/Generator/DocBlock/Tag/VarTag.php]: Variable name is allowed to be null in constructor, so there should be method to set it later. Other tags like `@param` or `@author` are working same way.

File: src/Generator/DocBlock/Tag/VarTag.php (hunk 2 above)
Comment (red-led) [inline src/Generator/DocBlock/Tag/VarTag.php]: done

File: src/Generator/DocBlock/Tag/VarTag.php (hunk 6 above)
Comment (red-led) [inline src/Generator/DocBlock/Tag/VarTag.php]: done

File: src/Generator/DocBlock/Tag/VarTag.php (hunk 7 above)
Comment (red-led) [inline src/Generator/DocBlock/Tag/VarTag.php]: done

File: src/Reflection/DocBlock/Tag/VarTag.php (hunk 9 above)
Comment (red-led) [inline src/Reflection/DocBlock/Tag/VarTag.php]: done

**Comment by red-led** [comment]
@Ocramius, updated

**Comment by Ocramius** [comment]
@red-led this is ready to go. I will still remove any premature mutability (setters) on merge, plus I'll need to rebase, since patch needs to be applied to `develop`. Expect a merge soon :)

**Comment by Ocramius** [comment]
@red-led thanks, merged manually into `develop` via zendframework/zend-code@cbdd65552c6f75e7a82b9a898f5e922ee64a0486

**Pull request closed without merging**